python/cudf/cudf/core/frame.py

```diff
@@ -1202,6 +1204,171 @@ def __arrow_array__(self, type=None):
+                "when upsampling the population `frac` > 1."
+            )
+        elif frac is not None and n is not None:
+            raise ValueError(
+                "Please enter a value for `frac` OR `n`, not both"
+            )
+
+        if frac is None and n is None:
+            n = 1
+        elif frac is not None:
+            if axis is None or axis == 0 or axis == "index":
+                n = int(round(self.shape[0] * frac))
+            else:
+                n = int(round(self.shape[1] * frac))
+
+        if axis is None or axis == 0 or axis == "index":
+            if not replace and n > self.shape[0]:
+                raise ValueError(
+                    "Cannot take a larger sample than population "
+                    "when 'replace=False'"
+                )
+
+            if weights is not None:
+                raise NotImplementedError(
+                    "weights is not yet supported for axis=0/index"
+                )
+
+            seed = (
+                random.randint(0, 10000)
+                if random_state is None
+                else np.int64(random_state)
+            )
+
+            result = self._from_table(
+                libcudf.copying.sample(
+                    self,
+                    n=n,
+                    replace=replace,
+                    seed=seed,
+                    keep_index=keep_index,
+                )
+            )
+            result._copy_categories(self)
+
+            return result
+        else:
+            if isinstance(self, cudf.Series):
+                raise ValueError(
+                    f"No axis named {axis} for "
+                    f"object type {self.__class__}"
+                )
+
+            columns = np.array(self._data.names)
+            if not replace and n > columns.size:
+                raise ValueError(
+                    "Cannot take a larger sample "
+                    "than population when 'replace=False'"
+                )
+
+            if weights is not None:
+                if is_column_like(weights):
+                    weights = np.array(weights)
+                else:
+                    raise ValueError(
+                        "Strings can only be passed to weights "
+                        "when sampling from rows on a DataFrame"
+                    )
+
+                if columns.size != len(weights):
+                    raise ValueError(
+                        "Weights and axis to be sampled must be of same length"
+                    )
+
+                total_weight = weights.sum()
+                if total_weight != 1:
+                    if not isinstance(weights.dtype, float):
+                        weights = weights.astype("float64")
+                    weights = weights / total_weight
+
+            np.random.seed(random_state)
+            gather_map = np.random.choice(
+                columns, size=n, replace=replace, p=weights
+            )
+
+            result = self[gather_map]
```

Comment:
This doesn't look safe to do on against a `Series` object

Reply:
Series doesn't support axis=1, and it will throw error.